WireDomain/Sources/WireDomain/Synchronization/IncrementalSync.swift

```diff
@@ -0,0 +1,169 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+import WireLogging
+
+public struct IncrementalSync: IncrementalSyncProtocol {
+
+    private let selfClientID: String
+    private let pushChannelAPI: any PushChannelAPI
+    private let updateEventsSync: any PullPendingUpdateEventsSyncProtocol
+    private let decryptor: any UpdateEventDecryptorProtocol
+    private let store: any UpdateEventsLocalStoreProtocol
+    private let processor: any UpdateEventProcessorProtocol
+    private let logger = WireLogger.sync
+
+    public init(
+        selfClientID: String,
+        pushChannelAPI: any PushChannelAPI,
+        updateEventsSync: any PullPendingUpdateEventsSyncProtocol,
+        decryptor: any UpdateEventDecryptorProtocol,
+        store: any UpdateEventsLocalStoreProtocol,
+        processor: any UpdateEventProcessorProtocol
+    ) {
+        self.selfClientID = selfClientID
+        self.pushChannelAPI = pushChannelAPI
+        self.updateEventsSync = updateEventsSync
+        self.decryptor = decryptor
+        self.store = store
+        self.processor = processor
+    }
+
+    public func perform() async throws -> Token {
+        logger.debug("performing incremental sync")
+        let pushChannel = try await pushChannelAPI.createPushChannel(clientID: selfClientID)
+        logger.debug("opening push channel")
+        let liveEventStream = try await pushChannel.open()
+        try await updateEventsSync.pull()
+        try await processStoredEvents()
+
+        let task = Task { @Sendable [logger, decryptor, store, processor] in
+            logger.debug("handling live event stream")
+
+            do {
+                for try await var envelope in liveEventStream {
+                    logger.debug("received live event envelope")
+
+                    // TODO: [WPB-16165] skip if duplicate.
+
+                    do {
+                        // Decrypt.
+                        logger.debug("decrypting live event envelope")
+                        envelope.events = try await decryptor.decryptEvents(in: envelope)
+                    } catch {
+                        logger.error("failed to decrypt live event envelope: \(String(describing: error))")
+                        continue
```

Comment:
thought: I wonder if we can split all this to have internal methods:

````
try await decrypt()
try await store()
try await process()
try await delete()
```

Reply:
Internal to the method scope or to the object? And for which goal? Improving readability?

Reply:
improving readability. I haven't thought much more. I usually find methods within methods weird so if i had to choose i would put it at the object level